File: Java/프로그래머스/1/468371. 노란불 신호등/노란불 신호등.java
import java.util.*;

class Solution {
    public int solution(int[][] signals) {
        int n = signals.length;
        int[] totalCycles = new int[n];
        
        for (int i = 0; i < n; i++) {
            totalCycles[i] = signals[i][0] + signals[i][1] + signals[i][2];
        }
        
        for (int t = 1; t <= 2000000; t++) {
            int yellowCount = 0;
            
            for (int i = 0; i < n; i++) {
                int g = signals[i][0];
                int y = signals[i][1];
                int cycle = totalCycles[i];
                
                int currentTime = t % cycle;
                if (currentTime == 0) currentTime = cycle;
                
                if (currentTime > g && currentTime <= (g+y)) {
                    yellowCount++;
                } else {
                    break;
                }
            }
            
            if (yellowCount == n) return t;
        }
        return -1;   
    }
}

File: Swift/프로그래머스/1/468371. 노란불 신호등/노란불 신호등.swift
import Foundation

func solution(_ signals:[[Int]]) -> Int {
    var cycles: [Int] = []
    for signal in signals {
        cycles.append(signal[0] + signal[1] + signal[2])
    }
    
    
    for t in 1...2000000 {
        var yCount = 0
        
        for i in 0..<signals.count {
            let g = signals[i][0]
            let y = signals[i][1]
            let cycle = cycles[i]
            
            var currentTime = t % cycle
            if currentTime == 0 { currentTime = cycle }
            
            if currentTime > g && currentTime <= g+y {
                yCount += 1
            } else {
                break
            }
        }
        
        if yCount == signals.count {
            return t
        }
    }
    return -1
}